src/Tribe/Views/V2/Views/By_Day_View.php

```diff
@@ -133,44 +135,45 @@ public function get_grid_days( $date = null, $force = false ) {
 		$this->warmup_cache( 'grid_days_found', 0, Cache_Listener::TRIGGER_SAVE_POST );
 		$events_per_day = $this->get_events_per_day();
 
+		// @todo @lucatume move this to the event repository?
+		Query::update_period_cache( $grid_start_date, $grid_end_date );
+		/** @var \Tribe__Cache $cache */
+		$cache = tribe('cache');
+
 		// phpcs:ignore
 		/** @var \DateTime $day */
 		foreach ( $days as $day ) {
 			$day_string = $day->format( 'Y-m-d' );
 
-			if ( isset( $this->grid_days_cache[ $day_string ] ) ) {
-				continue;
+			$cache_key   = Query::get_cache_key( $day_string );
+			$day_results = Events_Result_Set::from_value(
+				(array) $cache->get( $cache_key, Cache_Listener::TRIGGER_SAVE_POST )
+			);
+
+			if ( ! $day_results->count() ) {
```

Comment:
Probably personal preference here but I think moving empty state as "default" state of the ids removes the need of the else block¿ and negation operation

``` php
$event_ids = [];

if ( $day_results->count() ) {
	$day_results->order_by( $order_by, $order );
	$event_ids = array_map( 'absint', $day_results->pluck( 'ID' ) );
}
```

Reply:
Good suggestion, thanks.